src/Agent.Worker/Release/AgentUtilities.cs

```diff
@@ -3,30 +3,42 @@
 using System.Linq;
 using System.Text;
 
+using Microsoft.TeamFoundation.Core.WebApi;
+
 namespace Microsoft.VisualStudio.Services.Agent.Worker.Release
 {
     public static class AgentUtilities
     {
         // Move this to Agent.Common.Util
-        public static string GetPrintableEnvironmentVariables(IEnumerable<KeyValuePair<string, string>> variables)
+        public static string GetPrintableEnvironmentVariables(IExecutionContext executionContext)
         {
             StringBuilder builder = new StringBuilder();
-
-            if (variables != null)
+            
+            if (executionContext.Variables != null && executionContext.Variables.Public != null)
             {
-                var sortedVariables = variables.OrderBy(x => x.Key, StringComparer.OrdinalIgnoreCase);
+                IList<string> variablesToMask = GetVariablesToMaskForPublicProjects();
+                var sortedVariables = executionContext.Variables.Public.OrderBy(x => x.Key, StringComparer.OrdinalIgnoreCase);
                 foreach (var pair in sortedVariables)
                 {
-                    string varName = pair.Key.ToUpperInvariant().Replace(".", "_").Replace(" ", "_");
-                    builder.AppendFormat(
-                        "{0}\t\t\t\t[{1}] --> [{2}]",
-                        Environment.NewLine,
-                        varName,
-                        pair.Value);
+                    if (executionContext.Variables.System_TeamProjectVisibility != ProjectVisibility.Public
```

Comment:
for the email id variables, let's log the variable key as is. variable value can be the secret mask ********